apps/studio/src/components/CoreTabs.vue

```diff
@@ -1037,7 +1039,43 @@ import { TransportOpenTab, setFilters, matches, duplicate } from '@/common/trans
     },
     createQueryFromItem(item) {
       this.createQuery(item.text)
+    },
+    extractEntityName (q: string): string | null {
+      const query = q?.replace(/\s+/g, ' ').trim();
+
+      const patterns = [
+        { type: 'Stored Procedure', regex: /(create|alter)\s+procedure\s+(?:\[?\w+\]?\.)?\[?(\w+)\]?/i },
```

Comment:
Hey! I think just implementing this for table and table properties tabs would make sense. I don't think it makes sense in context of a query tab because a query can be for multiple entities.